R/linelist_merge.R

```diff
@@ -0,0 +1,416 @@
+      # handle this issue in next dictionary cleaning step
+    if(all(is.na(output_sheet$lab_result))){
+      print("lab_result")
+      output_sheet$lab_result<-NA
+    }
+
+    if(!("lab_resdate" %in% vars)){
+      print("lab_resdate")
+      output_sheet$lab_resdate<-NA
+      output_sheet$lab_resdate<-as.Date(output_sheet$lab_resdate,origin="1899-12-30")
+
+    }
+
+    #Recode this if missing use patcurrent status
+    if(!("patcourse_status" %in% vars)){
+      print("patcourse_status")
+      output_sheet$patcourse_status<-NA
+    }
+
+
+    #ensure this is for death, some files ie. KEN had date of outcome column whihc was used in this place
+    #some cases who were not dead would have this entred into their date of death
+    #make date of death missing if patcourse_status !=dead
+
+    if(!("patcourse_datedeath" %in% vars)){
+      print("patcourse_datedeath")
+      output_sheet$patcourse_datedeath<-NA
+      output_sheet$patcourse_datedeath<-as.Date(output_sheet$patcourse_datedeath,origin="1899-12-30")
+    }
+
+    if(!("patcurrent_status" %in% vars)) {
+      print("patcurrent_status")
+      output_sheet$patcurrent_status<-NA
+}
+
+if(length(grep("patsympt",vars))==0){
+        print("no patsmpt")
+        output_sheet$patsympt_other<-NA
+      }
+
+
+    if(!("patcourse_datedischarge" %in% vars)){
+      print("patcourse_datedischarge")
+      output_sheet$patcourse_datedischarge<-NA
+      output_sheet$patcourse_datedischarge<-as.Date(output_sheet$patcourse_datedischarge,origin="1899-12-30")
+    }
+
+
+    if(!("patinfo_ageonsetunit" %in% vars)){
+      print("patinfo_ageonsetunit")
+      output_sheet$patinfo_ageonsetunit<-NA
+    }
+
+    if(!("patinfo_ageonsetunitdays" %in% vars)){
+      print("patinfo_ageonsetunit")
+      output_sheet$patinfo_ageonsetunitdays<-NA
+    }
+
+
+
+    #keep variables of interest
+    output_sheet<- output_sheet %>% dplyr::select(report_date, patinfo_ageonset, patinfo_ageonsetunit,patinfo_ageonsetunitdays,
+                                                  patinfo_sex, patinfo_resadmin1,
+                                                  patinfo_resadmin2, report_classif,
+                                                  pat_symptomatic, pat_asymptomatic,
+                                                  comcond_preexist1, comcond_preexist,
+                                                  patinfo_occus, expo_travel, expo_travel_country,
+                                                  expo_contact_case, lab_result,
+                                                  lab_datetaken, lab_resdate,
+                                                  patcourse_status,
+                                                  patcourse_datedeath, patcourse_datedischarge, country_iso, contains("patsympt"), patcurrent_status)
+
+
+
+    #add cleaned output sheet to a list
+    output[[f]] <-output_sheet
+  }
+    #merge all cleaned sheets into one
```

Comment:
dplyr bind_rows

Reply:
Reduce is beneficial as doesn't matter about differing column classes. Avoids converting to column classes here and losing data (for example in the age column there is text written which is handled in the clean function)